code/modules/black_market/illegalradio.dm

```diff
@@ -0,0 +1,370 @@
+			return
+		new_listing.selected_description = desc
+		interact(usr)	
+		
+	else if(href_list["local_market_confirm_listing"])
+		if(money_stored < new_item_fee)
+			visible_message("\The [src] beeps: <span class='warning'>You actually don't have enough cash to list an item. That's pretty sad.</span>")	
+			return
+		else if(!new_listing || !new_listing.item)
+			visible_message("\The [src] beeps: <span class='warning'>There was an unexpected error! Try again later.</span>")
+			return
+		else if(!usr.Adjacent(new_listing.item))
+			visible_message("\The [src] beeps: <span class='warning'>Your item is too far away. You have to be near it.</span>")	
+			return
+		else
+			money_stored -= new_item_fee
+			player_market_items += new_listing
+			
+			var/obj/item/device/black_market_beacon/beacon = new /obj/item/device/black_market_beacon()
+			beacon.attach_to(new_listing.item, new_listing)
+			buzz_black_market()
+			
+			new_listing = null
+			open_screen(MAIN)
+		
+	..()
+
+
+/obj/item/device/illegalradio/proc/generate_main_menu(mob/user)
+	var/welcome = pick("We buzz you whenever the market changes!","Stop wasting bandwidth, buy something already!","Telecrystals ain't cheap, kid. Pay up.","There ain't nothing better than a good deal.","Back in my day, we didn't have 'teleporters'.","Human and Vox slaves NOT accepted as payment.","Absolutely no affiliation with Discount Dan.","Free from Nanotrasen regulation.","Too shy to come in person?","Unaffiliated with Spessmart(TM) supermarts.","Cluck, cluck. We've got no chickens.","What doth the greytide desire?","Wow, shooting spree? How original.","Uwa~ Senpai! Buy my stuff.","Japanese animes tolerated but condemned.","Goods from the TG1153 sector are prohibited. Go away.")
+
+	var/dat = list()
+	
+	//First, generate the header.
+	dat += "<B><font size=5>The Black Market</font></B><BR>"
+	dat += "<B>[welcome]</B><BR>"
+	dat += "Cash stored: [src.money_stored]<BR>"
+	dat += "<A href='byond://?src=\ref[src];dispense_change=1'>Eject Cash</a> <A href='byond://?src=\ref[src];toggle_notifications=1'>Buzz Notifications: [notifications ? "ON" : "OFF"]</a>"
+	dat += "<HR>"
+	dat += "<B>Request item:</B><BR>"
+	dat += "<I>Each item costs the price that follows its name. Cash only.</I><br><BR>"
+	var/list/buyable_items = get_black_market_items()
+	
+	//Next, create the player market entries at the top.
+	dat += "<b>Locally Sourced Goods</b><br>"
+	if(!player_market_items.len)
+		dat += "<font color='grey'><i>There are no items sourced from your station right now.</i></font><br>"
+	else
+		var/iterator = 0
+		for(var/datum/black_market_player_item/product in player_market_items)
+			iterator++
+			var/final_text = ""
+			if(product.selected_price <= money_stored)
+				final_text += "<A href='byond://?src=\ref[src];buy_local_item=[iterator];'>[product.selected_name]</A> ([product.selected_price])"
+			else
+				final_text += "<font color='grey'><i>[product.selected_name] ([product.selected_price]) </i></font>"
```

Comment:
Can you use a CSS class for this instead ?

Reply:
Do you mean nanoUI?

Reply:
He means any of our predefined span classes, you can even define your own. They do basically the same thing. CSS is just a way of saving you time. For example, instead of having to do font italics; font color = red every time, we just do span class 'warning'.